conan/tools/premake/premakedeps.py

```diff
@@ -84,34 +83,31 @@
 
 # Helper class that expands cpp_info meta information in lua readable string sequences
 class _PremakeTemplate(object):
-    def __init__(self, deps_cpp_info):
-        self.includedirs = ",\n".join('"%s"' % p.replace("\\", "/")
-                                      for p in
-                                      deps_cpp_info.includedirs) if deps_cpp_info.includedirs else ""
-        self.libdirs = ",\n".join('"%s"' % p.replace("\\", "/")
-                                  for p in deps_cpp_info.libdirs) if deps_cpp_info.libdirs else ""
-        self.bindirs = ",\n".join('"%s"' % p.replace("\\", "/")
-                                  for p in deps_cpp_info.bindirs) if deps_cpp_info.bindirs else ""
-        self.libs = ", ".join(
-            '"%s"' % p.replace('"', '\\"') for p in deps_cpp_info.libs) if deps_cpp_info.libs else ""
-        self.system_libs = ", ".join('"%s"' % p.replace('"', '\\"') for p in
-                                     deps_cpp_info.system_libs) if deps_cpp_info.system_libs else ""
-        self.defines = ", ".join('"%s"' % p.replace('"', '\\"') for p in
-                                 deps_cpp_info.defines) if deps_cpp_info.defines else ""
-        self.cxxflags = ", ".join(
-            '"%s"' % p for p in deps_cpp_info.cxxflags) if deps_cpp_info.cxxflags else ""
-        self.cflags = ", ".join(
-            '"%s"' % p for p in deps_cpp_info.cflags) if deps_cpp_info.cflags else ""
-        self.sharedlinkflags = ", ".join('"%s"' % p.replace('"', '\\"') for p in
-                                         deps_cpp_info.sharedlinkflags) \
-            if deps_cpp_info.sharedlinkflags else ""
-        self.exelinkflags = ", ".join('"%s"' % p.replace('"', '\\"') for p in
-                                      deps_cpp_info.exelinkflags) \
-            if deps_cpp_info.exelinkflags else ""
+    def __init__(self, dep_cpp_info, pkg_folder):
```

Comment:
This PremakeDeps refactor is using the ``package_folder`` directly in the generation, instead of modifying the dependencies ``cpp_info`` objects.